client/header/activity-panel/panels/inbox/action.js

```diff
@@ -0,0 +1,79 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { withDispatch } from '@wordpress/data';
+import PropTypes from 'prop-types';
+
+/**
+ * WooCommerce dependencies
+ */
+import { ADMIN_URL as adminUrl } from '@woocommerce/wc-admin-settings';
+
+/**
+ * Internal dependencies
+ */
+
+class InboxNoteAction extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			inAction: false,
+		};
+
+		this.handleActionClick = this.handleActionClick.bind( this );
+	}
+
+	handleActionClick( event ) {
+		const { action, noteId, triggerNoteAction } = this.props;
+		const href = event.target.href || '';
+		let inAction = true;
+
+		if ( href.length && ! href.startsWith( adminUrl ) ) {
+			event.preventDefault();
+			inAction = false; // link buttons shouldn't be "busy".
+			window.open( href, '_blank' );
+		}
+
+		this.setState( { inAction }, () => triggerNoteAction( noteId, action.id ) );
+	}
+
+	render() {
+		const { action } = this.props;
+		return (
+			<Button
+				isDefault
+				isPrimary={ action.primary }
+				isBusy={ this.state.inAction }
```

Comment:
Not directly related to this PR, but the busy state looks a bit weird:

<img width="236" alt="Screen Shot 2019-10-23 at 12 53 41 PM" src="https://user-images.githubusercontent.com/10561050/67359327-92014600-f595-11e9-9670-281b5a441988.png">